google/resource_kms_crypto_key_test.go

```diff
@@ -0,0 +1,373 @@
+package google
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestCryptoKeyIdParsing(t *testing.T) {
+	cases := map[string]struct {
+		ImportId            string
+		ExpectedError       bool
+		ExpectedTerraformId string
+		ExpectedCryptoKeyId string
+		Config              *Config
+	}{
+		"id is in project/location/keyRingName/CryptoKeyID format": {
+			ImportId:            "test-project/us-central1/test-key-ring/test-key-id",
+			ExpectedError:       false,
+			ExpectedTerraformId: "test-project/us-central1/test-key-ring/test-key-id",
+			ExpectedCryptoKeyId: "projects/test-project/locations/us-central1/keyRings/test-key-ring/cryptoKeys/test-key-id",
+		},
+		"id contains name that is longer than 63 characters": {
+			ImportId:      "test-project/us-central1/test-key-ring/can-you-believe-that-this-cryptokey-name-is-this-extravagantly-long",
+			ExpectedError: true,
+		},
+		"id is in location/keyRingName/CyptoKeyID format": {
+			ImportId:            "us-central1/test-key-ring/test-key-id",
+			ExpectedError:       false,
+			ExpectedTerraformId: "test-project/us-central1/test-key-ring/test-key-id",
+			ExpectedCryptoKeyId: "projects/test-project/locations/us-central1/keyRings/test-key-ring/cryptoKeys/test-key-id",
+			Config:              &Config{Project: "test-project"},
+		},
+		"id is in location/keyRingName/CyptoKeyID format without project in config": {
```

Comment:
For each of these test descriptions, could you use `cryptoKeyName` instead of `CryptoKeyID`?  This way, the format will be consistent with the error messages in `parseKmsCryptoKeyId`